src/notifications/notificationmanager.cpp

```diff
@@ -545,22 +545,15 @@ uint NotificationManager::handleNotify(int clientPid, const QString &appName, ui
 
     notification->restartProgressTimer();
 
-    const QString previewSummary(hints_.value(LipstickNotification::HINT_PREVIEW_SUMMARY).toString());
-    const QString previewBody(hints_.value(LipstickNotification::HINT_PREVIEW_BODY).toString());
-
+    // Use the summary and body as fallback values for previewSummary and previewBody.
+    // (This can be avoided by setting them explicitly to empty strings.)
+    if (!hints_.contains(LipstickNotification::HINT_PREVIEW_SUMMARY)) {
+        hints_.insert(LipstickNotification::HINT_PREVIEW_SUMMARY, summary);
+    }
+    if (!hints_.contains(LipstickNotification::HINT_PREVIEW_BODY)) {
+        hints_.insert(LipstickNotification::HINT_PREVIEW_BODY, body);
+    }
     if (androidOrigin) {
-        // If this notification includes a preview, ensure it has a non-empty body and summary
-        if (!previewSummary.isEmpty()) {
-            if (previewBody.isEmpty()) {
-                hints_.insert(LipstickNotification::HINT_PREVIEW_BODY, QStringLiteral(" "));
-            }
-        }
-        if (!previewBody.isEmpty()) {
-            if (previewSummary.isEmpty()) {
-                hints_.insert(LipstickNotification::HINT_PREVIEW_SUMMARY, QStringLiteral(" "));
-            }
-        }
```

Comment:
Wonder if there's any need for these in the apkd4 case.

Would be nice to skip the whole if(androidOrigin) block on newer version. Any reason not to?

Reply:
It's just about whether or not those priority overrides are still required, or whether apkd's derivation of that from the importance value coming from Android works for all those apps. I could find some way to guess if LipstickNotification::HINT_DISPLAY_ON should be set, then we can make all of that alien4 only I think.

But that requirement of needing both summary and body to be non-empty seems like something that shouldn't be needed these days anyway, for either alien version.

Reply:
> It's just about whether or not those priority overrides are still required

I think not. Better notification property support these days.

> I could find some way to guess if LipstickNotification::HINT_DISPLAY_ON should be set

Actually think there already is something for that. Then again I'm not sure should even that exist. For what I can tell Android doesn't do those display wake ups, thus the apps don't expect anything. Though some heuristic might not be too bad provided that it doesn't get applied too easily.